tests/test_utils_signal.py

```diff
@@ -63,8 +67,24 @@ def ok_handler(self, arg, handlers_called):
         reactor.callLater(0, d.callback, "OK")
         return d
 
-    def _get_result(self, signal, *a, **kw):
-        return send_catch_log_deferred(signal, *a, **kw)
+
+class SendCatchLogDeferredAsyncDefTest(SendCatchLogDeferredTest):
+
+    async def ok_handler(self, arg, handlers_called):
+        handlers_called.add(self.ok_handler)
+        assert arg == 'test'
+        await defer.succeed(42)
+        return "OK"
+
+
+@mark.only_asyncio()
+class SendCatchLogDeferredAsyncioTest(SendCatchLogDeferredTest):
+
+    async def ok_handler(self, arg, handlers_called):
+        handlers_called.add(self.ok_handler)
+        assert arg == 'test'
+        await asyncio.sleep(0.2)
+        return await get_from_asyncio_queue("OK")
```

Comment:
Could you please add integration-style test for this feature, maybe not to this module? Take some real signal which Scrapy is sending, implement async def callback for it, check that it works.

Reply:
I looked at the async-enabled signals and how are they used. There seem to be two groups.

`item_scraped`, `item_dropped`, `item_error`: these are related to `Scraper._itemproc_finished` but as far as I can see the result of that function is ignored and nothing waits for those signal handlers to finish, but I may be wrong.

`engine_started`, `engine_stopped`, `spider_opened`, `spider_closed`: these are included in the spider/engine startup/shutdown processes and the process waits for the handler to finish.

Reply:
That's actually a great information to document.